Sources/TuistGenerator/Linter/GraphLinter.swift

```diff
@@ -172,6 +173,32 @@ public class GraphLinter: GraphLinting {
         return dependencyIssues
     }
 
+    private func lintDuplicatedProductNamesInDependencies(graphTraverser: GraphTraversing) -> [LintingIssue] {
+        return graphTraverser.targets().flatMap { projectPath, projectTargets in
+            return projectTargets.flatMap { targetName, _ in
+                var seenProductNames: Set<String> = []
+                var duplicatedProductNames: Set<String> = []
+                for productName in graphTraverser
+                    .allTargetDependencies(path: projectPath, name: targetName)
+                    .map(\.target.productNameWithExtension)
+                {
```

Comment:
More of an open question, but do we know if this causes any performance hit? Each target loops always through all of its dependencies which, I'd think, can start to be expensive in large graphs. I'd consider memoising the individual subgraphs as we do in other places.

Reply:
I've added internal memoizing to the `allTargetDependencies` function.